lib/ui/recommendation.js

```diff
@@ -150,7 +154,7 @@ Recommendation.prototype = {
           this.el.appendChild(recommendation);
         }
         this.el.collapsed = false;
-        this.data = data;
+        this.data = payload;
```

Comment:
I'd prefer to keep `this.data` and get rid of `this.prev`. Is there any reason we need to keep two copies of the data?